ee/katc/sqlite.go

```diff
@@ -118,3 +132,34 @@ func querySqliteDb(ctx context.Context, slogger *slog.Logger, path string, query
 
 	return results, nil
 }
+
+// copySqliteDb makes a temporary directory and copies the given db into it.
+func copySqliteDb(path string) (string, error) {
+	dbCopyDir, err := agent.MkdirTemp("sqlite-temp")
```

Comment:
If we have to, we have to... Did we explore the sqlite options that might alleviate this?

Reply:
`mode=ro` wasn't making a difference -- let me try setting `immutable` to see!

Reply:
It works! Thank you, that's much tidier

Reply:
The docs say `database is opened read-only and all locking and change detection is disabled` so I was hopeful